src/__test__/components/plots-and-tables/ViolinPlotIndex.test.jsx

```diff
@@ -116,17 +117,20 @@ describe('ViolinIndex', () => {
 
     const newGeneShown = 'NewGeneShown';
     userEvent.type(geneInput, `{selectall}{del}${newGeneShown}`);
-    userEvent.click(rtl.getByRole(panelContainer, 'button'));
+    const searchButton = rtl.getByText(panelContainer, 'Search');
+    userEvent.click(searchButton);
+    await rtl.waitFor(() => expect(fetchCachedWork).toHaveBeenCalledTimes(3));
+
     expect(store.getState().componentConfig[plotUuid].config.shownGene).toBe(newGeneShown);
 
-    await rtl.waitFor(() => expect(generateSpecSpy).toHaveBeenCalledTimes(1));
+    await rtl.waitFor(() => expect(generateSpecSpy).toHaveBeenCalledTimes(3));
     await expectStringInVegaCanvas(newGeneShown, 1);
   });
 
   it('allows selection of the grouping/points, defaulting to louvain and All', async () => {
     await renderViolinIndex();
 
-    const dataSelection = rtl.screen.getAllByRole('tab')[1];
+    const dataSelection = rtl.screen.getAllByText('Select Data')[0];
     expect(dataSelection).toHaveTextContent('Select Data');
```

Comment:
Same like the comment above about `getAllByText('Select Data')`. If you're only looking for 1 element, you can just use `getByText('Select Data')`.

The replaced test checks for the order of the element. If this is not what we're testing for, instead of `toHaveTextContent`, you can just check for the existence of the element using something like `expect(dataSelection.length).toEqual(1)`